code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm

```diff
@@ -114,29 +119,39 @@ Doesn't work on other aliens/AI.*/
 	return
 
 /mob/living/carbon/alien/humanoid/proc/resin() // -- TLE
-	set name = "Secrete Resin (55)"
+	set name = "Secrete Resin"
 	set desc = "Secrete tough malleable resin."
 	set category = "Alien"
 
+	var/list/resin_buildings = list("Resin Wall (55)" = image(icon = 'icons/obj/smooth_structures/alien/resin_wall.dmi', icon_state = "resin_wall-0"),
+								"Resin Nest (55)" = image(icon = 'icons/mob/alien.dmi', icon_state = "nest"),
+								"Resin Door (80)" = image(icon = 'icons/obj/smooth_structures/alien/resin_door.dmi', icon_state = "resin"))
+	var/choice = show_radial_menu(src, src, resin_buildings, src, radius = 40)
 	if(powerc(55, TRUE))
-		var/choice = input("Choose what you wish to shape.","Resin building") as null|anything in list("resin wall","resin membrane","resin nest") //would do it through typesof but then the player choice would have the type path and we don't want the internal workings to be exposed ICly - Urist
-
-		if(!choice || !powerc(55, TRUE))
-			return
-		var/obj/structure/alien/resin/T = locate() in get_turf(src)
-		if(T)
+		var/built = FALSE
+		for(var/obj/structure/alien/resin/S in get_turf(src))
```

Comment:
can you make this use `locate()` instead?

Reply:
Donezo 👍